### PR TITLE
feat: Use 2 phase commit to solve DAO's lock period issue

### PR DESCRIPTION
**Pull request by xxuejie**
TODO:

* [x] phase 1 tests

### CODE REVIEW

File: c/dao.c [hunk 1]
```diff
@@ -328,6 +290,92 @@ static int calculate_dao_input_capacity(size_t input_index,
...
+  /* Check capacity */
+  uint64_t output_capacity = 0;
+  len = 8;
+  ret =
+      ckb_load_cell_by_field((unsigned char *)&output_capacity, &len, 0, index,
+                             CKB_SOURCE_OUTPUT, CKB_CELL_FIELD_CAPACITY);
+  if (ret != CKB_SUCCESS) {
+    return ret;
+  }
+  if (len != 8) {
```
[the remark this answers is not in the document]
Comment (xxuejie) [reply c/dao.c]: I'm not sure what it means to fill 8 bytes of data into its cells before depositing to DAO, the rule simply requires a newly created DAO cell to have 8 bytes of data when you are depositing to Nervous DAO

[the remark this answers is not in the document]
Comment (keroro520) [reply c/dao.c]: I know where I misunderstood... ignore it.

It requires that the inputs cells have `8` bytes output-data only for withdrawing(phase 1 and 2), but not depositing.

File: c/dao.c [hunk 2]
```diff
@@ -91,11 +46,11 @@
 #define EPOCH_LENGTH_MASK ((1 << EPOCH_LENGTH_BITS) - 1)
 
 /*
- * Fetch withdraw header hash from the last 8 bytes
+ * Fetch deposit header hash from the last 8 bytes
```
Comment (doitian) [inline c/dao.c]: outdated comment

Comment (xxuejie) [reply c/dao.c]: Done

**Comment by doitian** [comment]
Merge first. I'll keep reviewing it later.

**Pull request merged**